### DISCUSSION TITLE
add option for fast polling / slow polling for individual readings

### DISCUSSION THREAD

**Discussion by SkorpionMB** [emsesp/EMS-ESP32#1617] [Ideas]
An option for fast polling / slow polling for individual readings as it was on the km200 adapter would be useful
some readings as temperature or modulation would be useful to have more readings and other readings as modes settings enegy data dont need so often

**Comment by proddy** [comment]
EMS-ESP picks up telegrams as soon as they arrive on the double-decker bus (we call this broadcasted). Some are every 10 seconds, some 30 and some 60 seconds. We can force fetch some telegrams (we call fetch), but there's no guarantee they will be changed as it depends on how quickly the EMS client devices want to update. 

If you have specific examples of which telegrams and data points you're interested in mimicking the KM200, we can do some experimentation.

**Reply by SkorpionMB** [reply]
Hi
thank you for responding so fast 
in the old km200 adapter I set these datapoints for fast update ( 30sec )
*.actualTemp$, *.pumpModulation$, actualPower$, actualModulation$, *_t1$
*.actualTemp$ was used for more datapoints were the * can be anything
and these datapoints for slow update (twice a day )
*._Days$, *._Months, systemhealthStatus
all other datapoints had an update rate of 30 min
excep these i excluded from pollng completly as not needed
/gateway*, *switchPrograms*, *holidayModes*, *recordings*, *tank*, *DHW*, *tankLevel, info, secondBurner, *Pressure, brand, bus, systemType, *hs1.type, notifications

**Comment by MichaelDvP** [comment]
This is again a function of the ioBroker adapter, not a ems-esp function. Please go to https://github.com/tp1de/ioBroker.ems-esp or https://forum.iobroker.net/topic/45862/neuer-adapter-ems-esp-f%C3%BCr-bosch-heizungen
Afaik on the KM200 each value is fetched separate, but the adapter updates ems-esp per device. 
On start Thomas get `api/system/info` and extract the devices, then get `api/device/values` and extract the shortnames. Then (once) all entities are read separate (`api/<device>/<entity>`) to extract longname, uom, min/max, writeable, etc. and create the ioBroker entities. In normal operation only the `api/<device>` is called to update all device-values in one get. A fetch intervall per value is not usefull.
